File: frontend/src/app/shared/requests/chat-request.ts
import { ChatType } from '../models/chat-type.enum';
import { User } from '../models/user.model';
import { NotificationSetting } from '../models/notification-setting.model';

export interface ChatRequest {
  name: string;
  type: ChatType;
  createdById: string;
  organizationId: number;
  users: User[];
  usersSettings: NotificationSetting[];
}

File: backend/Watcher.Common/Requests/ChatRequest.cs
﻿namespace Watcher.Common.Requests
{
    using System.Collections.Generic;

    using Watcher.Common.Dtos;
    using Watcher.Common.Enums;

    public class ChatRequest
    {
        public ChatRequest() { }

        public string Name { get; set; }

        public ChatType Type { get; set; }
            
        public string CreatedById { get; set; }

        public int? OrganizationId { get; set; }

        public IList<UserDto> Users { get; set; }

        public IList<NotificationSettingRequest> UsersSettings { get; set; }
    }
}
